app/src/main/java/com/framgia/soundclound/screen/detailgenre/GenreDetailViewModel.java

```diff
@@ -0,0 +1,87 @@
+package com.framgia.soundclound.screen.detailgenre;
+
+import android.content.Context;
+import android.databinding.BaseObservable;
+import android.databinding.Bindable;
+import android.util.Log;
+import android.widget.Toast;
+
+import com.framgia.soundclound.BR;
+import com.framgia.soundclound.data.model.Track;
+import com.framgia.soundclound.data.source.TrackDataSource;
+import com.framgia.soundclound.data.source.TrackRepository;
+import com.framgia.soundclound.data.source.remote.TrackRemoteDataSource;
+import com.framgia.soundclound.util.Constant;
+
+import java.util.List;
+
+/**
+ * Created by Sony on 1/5/2018.
+ */
+
+public class GenreDetailViewModel extends BaseObservable implements TrackClickListener,
+        MoreInfoClickListener {
+
+    private GenreDetailAdapter mGenreDetailAdapter;
+    private String mGenre;
+    private Context mContext;
+
+    public GenreDetailViewModel(Context context, String genre) {
+        mContext = context;
+        mGenre = genre;
+        mGenreDetailAdapter = new GenreDetailAdapter();
+        mGenreDetailAdapter.setTrackClickListener(this);
+        mGenreDetailAdapter.setMoreInfoClickListener(this);
+        getData();
+    }
+
+    private void getData() {
+        TrackRepository trackRepository = new TrackRepository(TrackRemoteDataSource.getInstance());
+
+        trackRepository.getListTrack(Constant.BASE_URL + Constant.PARA,
+                mGenre, Constant.LIMIT_DEFAULT, Constant.OFF_SET_DEFAULT,
+                new TrackDataSource.Callback<List<Track>>() {
+                    @Override
+                    public void onStartLoading() {
+
+                    }
+
+                    @Override
+                    public void onGetSuccess(List<Track> data) {
+                        mGenreDetailAdapter.addData(data);
+                    }
+
+                    @Override
+                    public void onGetFailure(String message) {
+                        Toast.makeText(mContext , Constant.ERROR_TEXT, Toast.LENGTH_SHORT);
```

Comment:
WARNING: [ShowToast] Toast created but not shown: did you forget to call `show()` ?
 `Toast.makeText()` creates a `Toast` but does **not** show it. You must call `show()` on the resulting object to actually make the `Toast` appear.